cpp/src/parquet/column_reader.cc

```diff
@@ -1369,6 +1369,26 @@ class TypedRecordReader : public TypedColumnReaderImpl<DType>,
     return bytes_for_values;
   }
 
+  const void* ReadDictionary(int32_t* dictionary_length) override {
+    if (this->current_decoder_ == nullptr && !this->HasNextInternal()) {
+      dictionary_length = 0;
+      return nullptr;
+    }
+    // Verify the current data page is dictionary encoded. The current_encoding_ should
+    // have been set as RLE_DICTIONARY if the page encoding is RLE_DICTIONARY or
+    // PLAIN_DICTIONARY.
+    if (this->current_encoding_ != Encoding::RLE_DICTIONARY) {
+      std::stringstream ss;
+      ss << "Data page is not dictionary encoded. Encoding: "
```

Comment:
nit: should we also add `descr_->ToString()` to help debugging?

Reply:
maybe it's better to leave this as a user choice (they can still choose to print the descr using existing api)? I don't have a strong preference but let me know if you prefer always printing it.